fixes issue #128, long text not wrapping on /brigades or /projects

rkaufman13 opened the pull request:
Hi, this CSS fix should fix text wrapping on both `/brigades` and `/projects` as described in #128 

With all text wrapping, react-table defaults to table columns being equally wide, and the only way to set column widths appears to be using hardcoded pixel values (no percentages or flex-basis). That didn't look very good, so I also added a selector to target the "description" column and make it 40% of the table, which makes the other 3 columns share the remaining 60% equally. All this to say I'm totally happy if you wish to change the width of the description column to larger or smaller, but 40% looked fairly pleasing to my eye on a variety of monitor sizes.

k3KAW8Pnf7mkmdSMPHz27 commented on statusboard/src/components/ProjectsTable/ProjectsTable.scss:
```diff
@@ -16,11 +17,12 @@
     width: 100%;
     border-spacing: 0;
     border-radius: inherit;
+    table-layout:fixed;
```
I didn't know about `table-layout: fixed`, but it seems good for this purpose! Even more so if we end up adding better paging/infinite scrolling.

k3KAW8Pnf7mkmdSMPHz27 reviewed (commented):
> and the only way to set column widths appears to be using hardcoded pixel values (no percentages or flex-basis)

I am not sure I understand. Do you mean setting it from TS/JS instead of CSS? I think this is good, and it is reasonable to set it in `ProjectsTable.scss`.

I just have a few comments/questions, but otherwise, I'd go ahead and merge.

@mmazanec22 would you have time to take a look at this?

rkaufman13 commented:
Yes, we can hard-code column widths as a prop passed to each react-table
column, but the only values react-table accepts are pixels, rather than
percentages, so it wouldn't help in this case where we want to design for
multiple screen sizes :)
Table-layout:fixed is, I believe, required to set column widths that do not
resize based on the content, but I could be mistaken!

On Mon, Sep 6, 2021 at 4:43 PM Jonatan Asketorp ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> and the only way to set column widths appears to be using hardcoded pixel
> values (no percentages or flex-basis)
>
> I am not sure I understand. Do you mean setting it from TS/JS instead of
> CSS? I think this is good, and it is reasonable to set it in
> ProjectsTable.scss.
>
> I just have a few comments/questions, but otherwise, I'd go ahead and
> merge.
>
> @mmazanec22 <https://github.com/mmazanec22> would you have time to take a
> look at this?
> ------------------------------
>
> In statusboard/src/components/ProjectsTable/ColumnHeader.scss
> <https://github.com/codeforamerica/brigade-project-index-statusboard/pull/130#discussion_r703039302>
> :
>
> > @@ -11,6 +11,11 @@
>      margin-left: 1rem;
>    }
>
> +  input {
> +    width:100%;
> +    box-sizing: content-box;
>
> Isn't box-sizing helping with allowing margin between two columns? (or are
> you intentionally removing it)?
> ------------------------------
>
> In statusboard/src/components/ProjectsTable/ProjectsTable.scss
> <https://github.com/codeforamerica/brigade-project-index-statusboard/pull/130#discussion_r703064381>
> :
>
> > @@ -16,11 +17,12 @@
>      width: 100%;
>      border-spacing: 0;
>      border-radius: inherit;
> +    table-layout:fixed;
>
> I didn't know about table-layout: fixed, but it seems good for this
> purpose! Even more so if we end up adding better paging/infinite scrolling.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeforamerica/brigade-project-index-statusboard/pull/130#pullrequestreview-747390283>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEBHUR7EO3QUL53HPKDRBJ3UAURWPANCNFSM5DOZ2BNQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

k3KAW8Pnf7mkmdSMPHz27 commented:
> Yes, we can hard-code column widths as a prop passed to each react-table
column, but the only values react-table accepts are pixels, rather than
percentages, so it wouldn't help in this case where we want to design for
multiple screen sizes :)

That is unfortunate :/

> Table-layout:fixed is, I believe, required to set column widths that do not
resize based on the content, but I could be mistaken!

Even if we didn't need to set the column widths, it seems to be the right table layout. So I am definitively pro this change :)

The only thing I am not sure about is the change in margin for the header row,

**https://projects.brigade.network/**
<img width="428" alt="Skärmavbild 2021-09-08 kl  17 51 58" src="https://user-images.githubusercontent.com/2598631/132590787-b3428288-8299-4ef0-b2f7-6576a88f33fb.png">

**[Preview](https://deploy-preview-130--dreamy-snyder-8339ae.netlify.app/)**
<img width="428" alt="Skärmavbild 2021-09-08 kl  17 52 33" src="https://user-images.githubusercontent.com/2598631/132590756-465616dd-706a-44f7-a11c-dd640ba20f9b.png">

I like the extra space but it behaves slightly worse on small screens, so I'd be ok with keeping this change, I just want to make sure it is intentional. That is why I asked about the `box-margin` earlier. Except for that, this looks good to me and is good to go!

rkaufman13 commented:
Yes, so...it's intentional, but it's a tradeoff. I think the choice is
between headers being slightly less readable on small screens or the table
being slightly less readable on small screens. I personally think this is
the lesser of two evils but leave it up toyou :)

On Wed, Sep 8, 2021 at 5:59 PM Jonatan Asketorp ***@***.***>
wrote:

> Yes, we can hard-code column widths as a prop passed to each react-table
> column, but the only values react-table accepts are pixels, rather than
> percentages, so it wouldn't help in this case where we want to design for
> multiple screen sizes :)
>
> That is unfortunate :/
>
> Table-layout:fixed is, I believe, required to set column widths that do not
> resize based on the content, but I could be mistaken!
>
> Even if we didn't need to set the column widths, it seems to be the right
> table layout. So I am definitively pro this change :)
>
> The only thing I am not sure about is the change in margin for the header
> row,
>
> *https://projects.brigade.network/ <https://projects.brigade.network/>*
> [image: Skärmavbild 2021-09-08 kl 17 51 58]
> <https://user-images.githubusercontent.com/2598631/132590787-b3428288-8299-4ef0-b2f7-6576a88f33fb.png>
>
> *Preview <https://deploy-preview-130--dreamy-snyder-8339ae.netlify.app/>*
> [image: Skärmavbild 2021-09-08 kl 17 52 33]
> <https://user-images.githubusercontent.com/2598631/132590756-465616dd-706a-44f7-a11c-dd640ba20f9b.png>
>
> I like the extra space but it behaves slightly worse on small screens, so
> I'd be ok with keeping this change, I just want to make sure it is
> intentional. That is why I asked about the box-margin earlier. Except for
> that, this looks good to me and is good to go!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeforamerica/brigade-project-index-statusboard/pull/130#issuecomment-915601999>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEBHUR4Y7VA6AV4FV6QZH53UA7MEDANCNFSM5DOZ2BNQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

k3KAW8Pnf7mkmdSMPHz27 reviewed (approved):
> I think the choice is
between headers being slightly less readable on small screens or the table
being slightly less readable on small screens. I personally think this is
the lesser of two evils but leave it up to you :)

Ok. Well, in that case, this is good to go! Thank you for you contributions and sorry about the delay!
